### DISCUSSION TITLE
False positive for `reportUnboundVariable`

### DISCUSSION THREAD

**Discussion by injust** [microsoft/pyright#8978] [Q&A] [answered]
On L9 and L11, Pyright reports that `dest` is unbound. I don't understand how this is possible.

```python
cost = at = r = 0

while True:
    if at < 0:
        cost += r - 1
        at = r

    while at != (dest := at + 1):
        r = max(r, dest)
        cost += abs(at - dest)
        at = dest
```

[Pyright Playground](https://pyright-play.net/?code=MYewzgLgBAvFCG04CdZQAwChMHcAWAlgDYCmUAKsgK4kBcmUjUBAZgtADwb1O9SiQoAahRQAtFACMDPo0Rpk2XvmJl5AQjgAKACYlBtOPKFSAlD1lRUcALbwAHluQAaKHsimZsgdBEIARmBa8hLuEJ6W7GhhmEA)

**Comment by erictraut** [comment, marked as answer]
I think you meant to say "false positive". Yeah, I agree this is a bug.

I'll create an issue from the discussion.

**Reply by erictraut** [reply]
Here's a [link](https://github.com/microsoft/pyright/issues/8979) to the bug report for reference.
